pkg/transforms/policy.go

```diff
@@ -37,6 +41,121 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
 	if okns && okpp {
 		node.Properties["_parentPolicy"] = pnamespace + "/" + ppolicy
 	}
+
+	return &PolicyResource{node: node}
+}
+
+// For cert, config, operator policies.
+// This function returns `annotations`, `managedFields` for `source`,
+// and `severity`, `compliant`, `remediationAction`, `parentPolicyNs`, `parentPolicyName`
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	node.Properties["annotations"] = c.GetAnnotations()
+	node.Properties["isExternal"] = false
+
+	for _, m := range c.GetManagedFields() {
+		if m.Manager == "multicluster-operators-subscription" ||
+			strings.Contains(m.Manager, "argocd") {
+			node.Properties["isExternal"] = true
+		}
+	}
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	policyNsName, ok := c.GetLabels()["policy.open-cluster-management.io/policy"]
+	if ok {
+		stringSlice := strings.Split(policyNsName, ".")
+		// If len(stringSlice) != 2, failed to parse policyNsName
+		if len(stringSlice) == 2 {
+			node.Properties["parentPolicyNs"] = stringSlice[0]
+			node.Properties["parentPolicyName"] = stringSlice[1]
+		}
+	}
+
+	complianceState, found, err := unstructured.NestedString(c.Object, "status", "compliant")
+	if !found || err != nil {
+		node.Properties["compliant"] = ""
+	} else {
+		node.Properties["compliant"] = complianceState
+	}
+
+	severity, found, err := unstructured.NestedString(c.Object, "spec", "severity")
+	if !found || err != nil {
+		node.Properties["severity"] = ""
+	} else {
+		node.Properties["severity"] = severity
+	}
+
+	// remediationAction is same as `remediationAction`
+	remediationAction, found, err := unstructured.NestedString(c.Object, "spec", "remediationAction")
+	if !found || err != nil {
+		node.Properties["remediationAction"] = ""
+	} else {
+		node.Properties["remediationAction"] = remediationAction
+	}
+
+	return node
+}
+
+// For cert, config policies.
+func StandalonePolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	return &PolicyResource{node: getPolicyCommonProperties(c, node)}
+}
+
+func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	node = getPolicyCommonProperties(c, node)
+
+	var deploymentAvailable string
+	var upgradeAvailable string
+
+	conditions, found, err := unstructured.NestedSlice(c.Object, "status", "conditions")
+	if found && err == nil {
```

Comment:
Optional: You actually don't need this if statement since if it wasn't found or there was an error, `conditions` is an empty slice so the `for` loop will just not execute.